pinot-core/src/main/java/org/apache/pinot/core/function/scalar/SketchFunctions.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.function.scalar;
+
+import com.clearspring.analytics.stream.cardinality.HyperLogLog;
+import java.math.BigDecimal;
+import org.apache.datasketches.theta.Sketches;
+import org.apache.datasketches.theta.UpdateSketch;
+import org.apache.pinot.core.common.ObjectSerDeUtils;
+import org.apache.pinot.spi.annotations.ScalarFunction;
+import org.apache.pinot.spi.utils.CommonConstants;
+
+
+/**
+ * Inbuilt Sketch Transformation Functions
+ * The functions can be used as UDFs in Query when added in the FunctionRegistry.
+ * @ScalarFunction annotation is used with each method for the registration
+ *
+ * These are intended to be used during ingestion to create sketches from raw data, which can be rolled up later.
+ *
+ * Note this is defined in pinot-core rather than pinot-common because pinot-core has dependencies on
+ * datasketches/clearspring analytics.
+ *
+ * Example usage:
+ * {
+ *   "transformConfigs": [
+ *     {
+ *       "columnName": "players",
+ *       "transformFunction": "DistinctCountRawThetaSketch(playerID)"
+ *     },
+ *     {
+ *       "columnName": "names",
+ *       "transformFunction": "DistinctCountRawHLL(playerName)"
+ *     }
+ *   ]
+ * }
+ */
+public class SketchFunctions {
+  private SketchFunctions() {
+  }
+
+  /**
+   * Create a Theta Sketch containing the input
+   *
+   * @param input an Object we want to insert into the sketch, may be null to return an empty sketch
+   * @return serialized theta sketch as bytes
+   */
+  @ScalarFunction(nullableParameters = true)
+  public static byte[] distinctCountRawThetaSketch(Object input) {
+    // TODO make nominal entries configurable
+    UpdateSketch sketch =
+        Sketches.updateSketchBuilder().setNominalEntries(CommonConstants.Helix.DEFAULT_THETA_SKETCH_NOMINAL_ENTRIES)
```

Comment:
suggest make nominal entries configurable by adding 
```
  @ScalarFunction(nullableParameters = true)
  public static byte[] distinctCountRawThetaSketch(Object input) {
    distinctCountRawThetaSketch(input, DEFAULT_THETA_SKETCH_NOMINAL_ENTRIES);
  }
  
  @ScalarFunction(nullableParameters = true)
  public static byte[] distinctCountRawThetaSketch(Object input, int nominal_entries) { 
    ..
  }
```

Reply:
btw, is this entry event designed to be configurable? if we are always only adding 1 value to the sketch (same as HLL)

- this means we pre-determine the storage/accuracy trade-off during ingestion yes?

Reply:
1. yes, this will be pre-determining the storage/accuracy trade-off during ingestion
2. generally your `log2m` and `nominalEntries` should be the same throughout the system, since merging two sketches with different values will introduce a larger error margin, I believe this HyperLogLog library will throw an exception at merge time, but the Theta library will allow the merge

Reply:
also done, split so they can be configured, and updated examples of how to use it

Reply:
hmm, because these configs can be changed/modified after the table is created, it could lead to the error/exception you mentioned above. However, we still want to make this configurable per-column basis rather than relying on a table or cluster-wide setting.

several "solutions" here:
1. we can throw --> when merging 2 incompatible sketches we will not allow.
    - e.g. we only allow merge for the data ingested with the same config.
2. we can create table config validator to not allow changes to the sketches config afterwards
    - e.g. changes to this config after table creation is not allowed
3. we can regenerate all the derived columns when this config changes.

Reply:
1 is probably the most ideal solution here

I'm not sure 3 is viable because the usual reason you'd be generating sketches is to throw away the original data and compact rows down to save space